### PR TITLE
Implement Enumerable#{min_by, max_by} with count

### PR DESCRIPTION
**Pull request by bjfish**

### CODE REVIEW

File: src/main/ruby/core/enumerable.rb [hunk 1]
```diff
@@ -716,8 +716,23 @@ def max_n(n, &block)
...
+    n = Rubinius::Type.rb_num2long(n)
+    raise ArgumentError, "negative size #{n}" if n < 0
+    return [] if n == 0
+    pairs = []
+    each do
+      object = Truffle.single_block_arg
+      result = block.call(object)
+      pairs << [result, object]
+    end
+    pairs.sort_by {|p| p[0]}.reverse.first(n).map {|p| p[1]}
```
Comment (eregon) [inline src/main/ruby/core/enumerable.rb]: This implementation seems compatible, but it's quite inefficient for a small `n`.
What do MRI/JRuby do here?

Comment (bjfish) [reply src/main/ruby/core/enumerable.rb]: @eregon Fixed this to avoid creating the `pairs` array.

File: src/main/ruby/core/enumerable.rb [hunk 2]
```diff
@@ -716,8 +716,17 @@ def max_n(n, &block)
   end
```
Comment (eregon) [inline src/main/ruby/core/enumerable.rb]: Could you update `max_n` above to also use `last(n).reverse`?

File: src/main/ruby/core/enumerable.rb [hunk 3]
```diff
@@ -716,8 +716,17 @@ def max_n(n, &block)
...
   private :max_n
 
-  def max_by
+  def max_by_n(n, &block)
+    n = Rubinius::Type.rb_num2long(n)
+    raise ArgumentError, "negative size #{n}" if n < 0
+    return [] if n == 0
+    self.sort_by(&block).last(n).reverse
+  end
+  private :max_by_n
```
Comment (eregon) [inline src/main/ruby/core/enumerable.rb]: I personally prefer `private def` as it clarifies it's a helper method from the start.

Comment (bjfish) [reply src/main/ruby/core/enumerable.rb]: Okay, updated to `private def`

File: src/main/ruby/core/enumerable.rb (hunk 2 above)
Comment (bjfish) [reply src/main/ruby/core/enumerable.rb]: Yes, updated `max_n` to `last(n).reverse`

**Pull request merged**